contracts/ArchetypeLogic.sol

```diff
@@ -328,12 +334,13 @@ library ArchetypeLogic {
     mapping(address => OwnerBalance) storage _ownerBalance,
     mapping(address => mapping(address => uint128)) storage _affiliateBalance,
     address affiliate,
-    uint256 quantity
+    uint256 quantity,
+    uint256 curSupply
   ) public {
     address tokenAddress = i.tokenAddress;
     uint128 value = uint128(msg.value);
     if (tokenAddress != address(0)) {
-      value = uint128(computePrice(i, config.discounts, quantity, affiliate != address(0)));
+      value = uint128(computePrice(i, config.discounts, quantity, curSupply, affiliate != address(0)));
```

Comment:
need to add a test to archetype.test.ts to confirm that this value matches expectations.